File: python/7 kyu/299.py
# https://www.codewars.com/kata/5b097da6c3323ac067000036/train/python

# Passed

def solve(a, b):
    a_points, b_points = 0, 0
    for a_point, b_point in zip(a, b):
        if a_point > b_point:
            a_points += 1
        elif b_point > a_point:
            b_points += 1

    if a_points > b_points:
        result = f'{a_points}, {b_points}: Alice made "Kurt" proud!'
    elif b_points > a_points:
        result = f'{a_points}, {b_points}: Bob made "Jeff" proud!'
    else:
        result = f'{a_points}, {b_points}: that looks like a "draw"! Rock on!'
        
    return result

output = solve([25, 50, 22], [34, 49, 50])
print(output)

File: nodejs/7 kyu/299.js
// https://www.codewars.com/kata/5d376cdc9bcee7001fcb84c0/train/javascript

// Passed

function oddOnesOut(nums) {
  let numToCountMap = new Object();

  for (const num of nums) {
    if (!(num in numToCountMap)) {
      numToCountMap[num] = 0;
    }

    numToCountMap[num]++;
  }

  let result = nums.filter((num) => numToCountMap[num] % 2 === 0);
  return result;
}

const output = oddOnesOut([
  82, 86, 71, 58, 44, 79, 50, 44, 79, 67, 82, 82, 55, 50,
]);
console.log(output);